### ISSUE TITLE
Is there a way how to add Eurotronic Comet Z-Wave Energy saving radiator support?

### ISSUE THREAD

**Issue by davidrybar** [Mariano-Github/Edge-Drivers-Beta#68]
I have exchanged couple emails with Eurotronic support, but without any result. The device registers as Dimmer Switch in SmartThings and there seem no suitable driver alternative to make it work. 

It's https://eurotronic.org/wp-content/uploads/2023/04/Comet-Z-Wave-EN-2023-04-06.pdf

Thank you for the answer.

**Comment by Mariano-Github** [comment]
Hi
Are you try installing this Eurotronics driver?

https://zwave.eu/smartthings_edge_driver_collection/

**Comment by davidrybar** [comment]
Hi!
Thank your for such fast response. I have installed the drivers from zwave.eu link you have mentioned:

![image](https://user-images.githubusercontent.com/34548098/235248522-56d58f6b-137e-4702-af50-58ceccef6640.png)

However, I ended with same result:

![image](https://user-images.githubusercontent.com/34548098/235248780-c9f93d4f-bc68-4b2b-a354-49d61cb764d2.png)

Does it mean it's paired but driver not recognized? I'm new in this area so maybe I miss something.

**Comment by davidrybar** [comment]
I suspect that Spirit Radiator Thermostat is different product compared to Eurotronic COMET Z-Wave Plus Thermostat

**Comment by Mariano-Github** [comment]
You can see in the menu three points, driver option, with which driver has it been paired?

**Comment by davidrybar** [comment]
I've already checked that, because I changed the driver for another device - siren. But in this case, in three points menu, there are just options edit (name, location, room) and Information (title Dimmer Switch, Controller version 2.3.17-11 and link for Open source licences). There is no way how to manipulate with driver options :(

**Comment by davidrybar** [comment]
It seems the device is this one: https://products.z-wavealliance.org/products/1601?selectedFrequencyId=1

However, not recognized by any driver yet.

**Comment by Mariano-Github** [comment]
If device is paired with a edge driver the setting menu is same this
![Screenshot_20230428-233005](https://user-images.githubusercontent.com/74271621/235258057-8baa9ad0-d0c9-43ec-ab21-b0b0b0aa4411.png)

You can install Z-wave Thing Mc driver to see the device info

**Comment by davidrybar** [comment]
I have this option other devices, but not in this Thermostat aka Dimmer:

![image](https://user-images.githubusercontent.com/34548098/235260194-821a5ca6-5584-4d62-ab0d-0b3f65b506d1.png)

**Comment by davidrybar** [comment]
I'm using IPhone app

**Comment by davidrybar** [comment]
I have installed Z-Wave Thing Mc, but it stills pairs as Dimmer
![image](https://user-images.githubusercontent.com/34548098/235265064-5da40df9-14a7-4830-a47d-3793c379bb7d.png)

thank you very much for your time and help

**Comment by Mariano-Github** [comment]
You have to delete the DTH it pairs with in IDE

**Comment by Mariano-Github** [comment]
I'll see tomorrow, I'm going to sleep

**Comment by Mariano-Github** [comment]
Hi,

The only thing I can do is add the fingerprints to the Z-Wave Thermostat Mc driver to see if it works with the default libraries.

I guess Eurotronics will add to your driver

```
───────────────────────────────────────────────────
 Driver Id    ae1317ad-ba10-4f8b-9926-986101f2b849 
 Name         Z-Wave Thermostat Mc
 Version      2023-04-29T08:53:35.132577234        
───────────────────────────────────────────────────
- id: Eurotronic Comet/Radiator
    deviceLabel: Eurotronic Comet Radiator
    manufacturerId: 0x0148
    productType: 0x0002
    productId: 0x0001
    deviceProfileName: base-radiator-thermostat
```

**Comment by davidrybar** [comment]
Thanks a lot, I will test it, maybe it will work.

I have contacted Eurotronic support week ago, but it looks like it's not priority for them. They stopped responding to Homey integration (I was testing this device with Homey Bridge too with same result like SmartThins)

As a developer, I would develop and test the driver by myself, but whole ecosystem is new to me so I'm lost a little bit.

Thank you for your time!

**Comment by davidrybar** [comment]
Mariano,
I was able to write own driver with lua based on base-radiator-thermostat and pair the device with SmartThings

![image](https://user-images.githubusercontent.com/34548098/235426564-21d37b16-af99-43f8-a604-942b8720a023.png)

It works, however, I did not implement child lock and energy save heating functionality yet. I'm not happy how the ecosystem works (pairing) and not sure if it's because of hub, thermostat, software or z-wave itself.

Anyway, thank you for your responses - it helped me to make it work even though I would expect Eurotronic to do this job.

The correct fingerprint is:
    manufacturerId: 0x0148
    productType: 0x0004
    productId: 0x0003
